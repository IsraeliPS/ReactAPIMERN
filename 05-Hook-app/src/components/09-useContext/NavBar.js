import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
  const activeStyle = {
    textDecoration: 'underline',
    color: 'white'
  }
  return (

    <nav className='navbar navbar-expand-lg navbar-dark bg-dark'>

      <Link className='navbar-brand' to='/'>useContext</Link>

      <NavLink
        style={({ isActive }) =>
          isActive ? activeStyle : undefined} className='nav-item nav-link' aria-current='page' to='/'
      >Home
      </NavLink>
      <NavLink className='nav-item nav-link' to='/login'>Login</NavLink>
      <NavLink className='nav-item nav-link' to='/about'>About</NavLink>

    </nav>
  )
}
