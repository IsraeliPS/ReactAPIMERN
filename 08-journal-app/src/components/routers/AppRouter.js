
import React from 'react'
import {
  BrowserRouter as Router,
  Switch,
  Redirect
  , Route
} from 'react-router-dom'
import { JournalScreen } from '../journal/JournalScreen'

// import { useDispatch } from 'react-redux'

// import { firebase } from '../firebase/firebase-config'
import { AuthRouter } from './AuthRouter'
// import { PrivateRoute } from './PrivateRoute'


// import { login } from '../actions/auth'
// import { PublicRoute } from './PublicRoute'
// import { startLoadingNotes } from '../actions/notes'

export const AppRouter = () => {
  // const dispatch = useDispatch()

  // const [checking, setChecking] = useState(true)
  // const [isLoggedIn, setIsLoggedIn] = useState(false)

  // useEffect(() => {
  //   firebase.auth().onAuthStateChanged(async (user) => {
  //     if (user?.uid) {
  //       dispatch(login(user.uid, user.displayName))
  //       setIsLoggedIn(true)
  //       dispatch(startLoadingNotes(user.uid))
  //     } else {
  //       setIsLoggedIn(false)
  //     }

  //     setChecking(false)
  //   })
  // }, [dispatch, setChecking, setIsLoggedIn])

  // if (checking) {
  //   return (
  //     <h1>Wait...</h1>
  //   )
  // }
  return (
    <Router>
      <div>
        <Switch>
          <Route
            path='/auth'
            component={AuthRouter}
          />
          <Route
            exact
            path='/'
            component={JournalScreen}
          />
          {/* <PublicRoute
            path='/auth'
            component={AuthRouter}
            isAuthenticated={isLoggedIn}
          />

          <PrivateRoute
            exact
            isAuthenticated={isLoggedIn}
            path='/'
            component={JournalScreen}
          /> */}

          <Redirect to='/auth/login' />

        </Switch>
      </div>
    </Router>
  )
}
