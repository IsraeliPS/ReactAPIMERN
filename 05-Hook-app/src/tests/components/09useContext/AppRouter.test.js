import React from 'react'
import { AppRouter } from '../../../components/09-useContext/AppRouter'
import { mount } from 'enzyme'

describe('Pruebas en <AppRouter />', () => {
  const wrapper = mount(
    <AppRouter />
  )

  test('Debe de mostrarse correctamente', () => {

  })
})
