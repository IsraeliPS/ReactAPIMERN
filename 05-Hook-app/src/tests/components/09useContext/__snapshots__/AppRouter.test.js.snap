// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe de mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      useContext
    </a>
    <a
      aria-current="page"
      className="nav-item nav-link active"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "color": "white",
          "textDecoration": "underline",
        }
      }
    >
      Home
    </a>
    <a
      className="nav-item nav-link"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
    <a
      className="nav-item nav-link"
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre />
    </div>
  </div>
</div>
`;
